zulipterminal/ui_tools/boxes.py

```diff
@@ -1186,7 +1186,21 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.model.controller.view.middle_column.set_focus('footer')
         elif is_command_key('QUOTE_REPLY', key):
             self.keypress(size, 'enter')
-            quote = '```quote\n' + self.model.client.get_raw_message(
+
+            # Compose box should look something like this:
+            #   @_**Zeeshan|514** [said](link to message):
+            #   ```quote
+            #   message_content
+            #   ```
+            quote = '@_**{}|{}** '.format(self.message['sender_full_name'],
+                                          self.message['sender_id'])
+
+            # FIXME: The type and generation of link needs to be discussed.
+            # For interoperability webapp link should be given, but
+            # that would not narrow to the quoted message inside ZT.
+            absolute_url = absolute_url_of_message(self.model, self.message)
+            quote += '[said]({}):\n'.format(absolute_url)
+            quote += '```quote\n' + self.model.client.get_raw_message(
```

Comment:
This will fail to correctly quote a message that already contains a quote or a code block. You need to quote ```` ``` ```` with ````` ```` `````, ````` ```` ````` with `````` ````` ``````, and so on. See [`get_unused_fence`](https://github.com/zulip/zulip/blob/160cc5120aae0870c68e6e3a12a391035bd25843/static/shared/js/fenced_code.js#L241-L251) (called [here](https://github.com/zulip/zulip/blob/160cc5120aae0870c68e6e3a12a391035bd25843/static/js/compose_actions.js#L444)) in the web app.

Reply:
Thanks for the review @andersk. I probably missed that part of code logic, from the webapp. Nevertheless, I am now supposed to refactor the whole code along with some server code into a new module `server_urls.py` as instructed by Neil in CZO. I'll keep this review in mind while refactoring.